backbone.js

```diff
@@ -1079,10 +1079,26 @@
       if (!wait) this.add(model, options);
       var collection = this;
       var success = options.success;
+      var forwardPristineError;
       options.success = function(m, resp, callbackOpts) {
-        if (wait) collection.add(m, callbackOpts);
+        if (wait) {
+          model.off('error', forwardPristineError);
+          collection.add(m, callbackOpts);
+        }
```

Comment:
However, this introduces a subtle bug if the model is, indeed, already in the collection. In that case, `forwardPristineError` is `undefined` and these lines will remove *all* `'error'` listeners from `model`. To fix that, I could repeat the `this.has` check here, but that is expensive, *or*